python/thunder/rdds/fileio/readers.py

```diff
@@ -438,23 +432,16 @@ def __getMatchingKeys(self, dataPath, filename=None):
                         keyName = ""
             keyName += filename
 
-        return _BotoS3Client.retrieveKeys(bucket, keyName, prefix=parse[2], postfix=parse[3])
+        return _BotoS3Client.retrieveKeys(bucket, keyName, prefix=parse[2], postfix=parse[3],
+                                          excludeDirectories=excludeDirectories, recursive=recursive)
 
-    def list(self, dataPath, filename=None, startIdx=None, stopIdx=None, recursive=False, includeDirectories=True):
+    def list(self, dataPath, filename=None, startIdx=None, stopIdx=None, recursive=False, includeDirectories=False):
         """List s3 objects specified by dataPath.
 
         Returns sorted list of 's3n://' URIs.
         """
-        # TODO: note that the default value for includeDirectories is here True, which reflects the current (and
-        # longstanding) behavior of this class. This *differs* from the local FS list() method, which by default
-        # filters out directories (which is definitely what we want if we're reading with recursive=True).
-        # These two need to be made more consistent.
```

Comment:
This TODO comment appears to have been inaccurate. This `list` function seems to have been filtering directory listings out of the returned results, regardless of the `includeDirectories` setting. I changed the default value here to False to better reflect the actual behavior, no matter what this comment claims, and made it so that if `includeDirectories=True` then directories will be returned (provided that there are actual S3 objects corresponding to the directory, and not just an implicit directory by the convention of using '/' as a path delimiter on S3).